deploy/gk-deploy

```diff
@@ -515,14 +508,37 @@ if [[ "${EXECUTOR}" == "ssh" ]]; then
   done <<< "$(echo -e "${NODES}")"
 fi
 
-if [[ ${LOAD} -eq 0 ]]; then
-  output -n "Checking that heketi pod is not running ... "
-  check_pods "glusterfs=heketi-pod" 2
-  if [[ $? -eq 0 ]]; then
-    output "Found heketi pod running. Please destroy existing setup and try again."
-    exit 1
-  fi
-  output "OK"
+output "Checking for pre-existing resources..."
+
+output -n "  GlusterFS pods ... "
+check_pods "glusterfs=pod" 2 2>&1
+if [[ $? -eq 0 ]]; then
+  EXISTS_GLUSTERFS=1
+  output "found."
+else
+  output "not found."
+fi
+
+output -n "  deploy-heketi pod ... "
+check_pods "deploy-heketi=pod" 2 2>&1
+if [[ $? -eq 0 ]]; then
+  EXISTS_DEPLOY_HEKETI=1
+  output "found."
+else
+  output "not found."
+fi
+
+output -n "  heketi pod ... "
+check_pods "heketi=pod" 2 2>&1
+if [[ $? -eq 0 ]]; then
+  EXISTS_HEKETI=1
+  output "found."
+else
+  output "not found."
+fi
+
+if [[ ${EXISTS_HEKETI} -eq 0 ]]; then
+  output -n "Creating initial resources ... "
   if [[ "${CLI}" == *oc\ * ]]; then
     eval_output "${CLI} create -f ${TEMPLATES}/deploy-heketi-template.yaml 2>&1"
```

Comment:
Two reasons.

1. It is possible that we have already successfully deployed deploy-heketi, completed the setup-openshift-heketi-storage command, deleted the deploy-heketi resources, and then failed or exited the script somehow. Since everything could be already setup for the final heketi pod, we should make sure these are always present.

2. This is looking ahead to the possibility of allowing the option to use a lube secret foe the heketi db, and thus would not need deploy-heketi